### ISSUE TITLE
POST with new relation

### ISSUE THREAD

**Issue by Sybio** [api-platform/api-platform#371]
I would like to know if it is possible to POST with a new relation, like this : 

```json
# POST /picks
{
    "bookmaker": "/bookmakers/6ee03de1-711a-11e7-9be1-848f69ce56ef",
    "analysis": "This is my text.",
    "stake": 35.5,
    "confidence": 6,
    "type": "single",
    "subpicks": [
    	{
        	"@type": "SubpickTypeFree",
    	        "bet": "My custom bet!",
    	        "odd": 1.5,
    		"dateKickOff": "2017-07-30T20:00:00+00:00",
    		"category": "/categories/6ee84bf5-711a-11e7-9be1-848f69ce56ef"
	}
    ]
}
```

Here I have an entity **Pick** having a OneToMany relation (property "$subpicks") with **Subpick** and **I would like to POST a new Pick, with a new Subpick at the same time**.

When I perform the action, i got this error : 
```
Nested documents for attribute "subpicks" are not allowed. Use IRIs instead.
```
I also noticed that using an IRI ```/subpicks``` instead of ```subpicks``` doesn't create subpicks, but doesn't display error (Pick is created, without subpicks, attribute */subpicks* is ignored) : 

```json
# POST /picks
{
    "bookmaker": "/bookmakers/6ee03de1-711a-11e7-9be1-848f69ce56ef",
    # ...
    "/subpicks": [
    	{
        	"@type": "SubpickTypeFree",
    	        "bet": "My custom bet!",
    	        "odd": 1.5,
    		"dateKickOff": "2017-07-30T20:00:00+00:00",
    		"category": "/categories/6ee84bf5-711a-11e7-9be1-848f69ce56ef"
	}
    ]
}
```

To finish, I also noticed that pointing existing relation works, it will modify (like a PUT) the subpick *pick_id* field by the new Pick ID : 

```json
# POST /picks
{
    "bookmaker": "/bookmakers/6ee03de1-711a-11e7-9be1-848f69ce56ef",
    # ...
    "subpicks": [
    	"/subpicks/6ff553b4-711a-11e7-9be1-848f69ce56ef"
    ]
}
```

So is there a solution to create resource and relations at the same time ?

I'm close to giving up, any help could be great !

**Comment by Gounlaf** [comment]
Hi @Sybio,

**TL;DR** [Take a look at the documentation (search Person / Book)](https://api-platform.com/docs/core/serialization-groups-and-relations#header)

If I'm right, you have to use `@Groups` (documented here: https://api-platform.com/docs/core/serialization-groups-and-relations#header)

* on your **Pick** entity, define groups for `normalization` and `denormalization`.
* on your **Subpick** entity, use defined groups on the properties (in `@Groups`, you will have groups for the **Subpick** entity entity itself, and for the **Pick** entity entity too)

**Comment by soyuka** [comment]
And you may need `cascade: persist` on the relation :).

**Comment by Sybio** [comment]
Ok I just forgot the denormalization_context... Thanks for your help !

**Comment by dawkaa** [comment]
> Ok I just forgot the denormalization_context... Thanks for your help !

where do this the denormalization_context add ? plss

**Comment by Gounlaf** [comment]
> > Ok I just forgot the denormalization_context... Thanks for your help !
> 
> where do this the denormalization_context add ? plss any code could be great

https://api-platform.com/docs/core/subresources/#using-serialization-groups
